pkg/graph/schema.graphql

```diff
@@ -665,6 +666,51 @@ type SystemIntake {
   updatedAt: Time!
 }
 
+"""
+Indicates which action should be taken
+"""
+enum SystemIntakeActionType {
+  BIZ_CASE_NEEDS_CHANGES
+  CREATE_BIZ_CASE
+  GUIDE_RECEIVED_CLOSE
+  ISSUE_LCID
+  NEED_BIZ_CASE
+  NO_GOVERNANCE_NEEDED
+  NOT_IT_REQUEST
+  NOT_RESPONDING_CLOSE
+  PROVIDE_FEEDBACK_NEED_BIZ_CASE
+  PROVIDE_GRT_FEEDBACK_BIZ_CASE_DRAFT
+  PROVIDE_GRT_FEEDBACK_BIZ_CASE_FINAL
+  READY_FOR_GRB
+  READY_FOR_GRT
+  REJECT
+  SEND_EMAIL
+  SUBMIT_BIZ_CASE
+  SUBMIT_FINAL_BIZ_CASE
+  SUBMIT_INTAKE
+}
+
+"""
+An action taken on a system intake, often resulting in a change in status.
+"""
+type SystemIntakeAction {
+  id: UUID!
+  systemIntake: SystemIntake!
```

Comment:
Does this end up with a circular reference? Or do I remember that somehow GQL has a protection against this built in...?

Reply:
Great question. I had to look this up, and here is where I ended up: It is a circular reference, but it should be OK for a few reasons:

* The client has to specify the entire query, so if they wanted to go 10 levels deep here, the query would have to be ten levels deep as well. The fact that you have to explicitly list fields in GraphQL works in our favor here.
* We own the only client, so we can be sure not to do that.
* gqlgen has a concept of [query complexity](https://gqlgen.com/reference/complexity/) that we can use to limit how expensive the queries that we handle can be. I haven't looked into this that much yet, but my thought was that we should start logging the complexity for queries to get a sense of what is normal. Then we can figure out if we should add a limit or how to handle scenarios where queries become too expensive.